test_vae_training/2_check_trained_model.py

```diff
@@ -0,0 +1,214 @@
+
+# Read in config variables
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "test_vae_training", "config_current_vae.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+dataset_name = params["dataset_name"]
+NN_architecture = params["NN_architecture"]
+normalized_compendium_filename = params["normalized_compendium_filename"]
+
+NN_dir = os.path.join(base_dir, dataset_name, "models", NN_architecture)
+# -
+
+# Import normalized data
+normalized_compendium = pd.read_csv(
+    normalized_compendium_filename, sep="\t", index_col=0, header=0
+)
+
+normalized_compendium.head()
+
+# Drop outliers found manually
+samples_to_drop = [
+    "PRJEB34610_ERR3561806",
+    "PRJEB34610_ERR3561830",
+    "PRJNA297268_SRR2568180",
+]
+normalized_compendium = normalized_compendium.drop(samples_to_drop)
+
+# ### Plot umap of normalized data
+
+pca = PCA(n_components=2)
+
+# +
+# Get and save PCA model
+model = pca.fit(normalized_compendium)
+# model = umap.UMAP(random_state=random_state).fit(normalized_compendium)
+
+compendium_PCAencoded = model.transform(normalized_compendium)
+
+compendium_PCAencoded_df = pd.DataFrame(
+    data=compendium_PCAencoded, index=normalized_compendium.index, columns=["1", "2"]
+)
+
+# +
+# Plot
+fig = ggplot(compendium_PCAencoded_df, aes(x="1", y="2"))
+fig += geom_point(alpha=0.2)
+fig += labs(x="PCA 1", y="PCA 2", title="PCA normalized compendium")
+fig += theme_bw()
+fig += theme(
+    legend_title_align="center",
+    plot_background=element_rect(fill="white"),
+    legend_key=element_rect(fill="white", colour="white"),
+    legend_title=element_text(family="sans-serif", size=15),
+    legend_text=element_text(family="sans-serif", size=12),
+    plot_title=element_text(family="sans-serif", size=15),
+    axis_text=element_text(family="sans-serif", size=12),
+    axis_title=element_text(family="sans-serif", size=15),
+)
+fig += guides(colour=guide_legend(override_aes={"alpha": 1}))
+
+print(fig)
+# -
+
+# ### Plot encoded data
+
+# +
+# Load VAE models
+model_encoder_file = glob.glob(os.path.join(NN_dir, "*_encoder_model.h5"))[0]
+
+weights_encoder_file = glob.glob(os.path.join(NN_dir, "*_encoder_weights.h5"))[0]
+
+model_decoder_file = glob.glob(os.path.join(NN_dir, "*_decoder_model.h5"))[0]
+
+weights_decoder_file = glob.glob(os.path.join(NN_dir, "*_decoder_weights.h5"))[0]
+
+# Load saved models
+loaded_model = load_model(model_encoder_file)
+loaded_decode_model = load_model(model_decoder_file)
+
+loaded_model.load_weights(weights_encoder_file)
+loaded_decode_model.load_weights(weights_decoder_file)
+
+# +
+# Encode normalized compendium into latent space
+compendium_encoded = loaded_model.predict_on_batch(normalized_compendium)
+
+compendium_encoded_df = pd.DataFrame(
+    data=compendium_encoded, index=normalized_compendium.index
+)
+
+# +
+# Get and save PCA model
+model = pca.fit(compendium_encoded_df)
+# model = umap.UMAP(random_state=random_state).fit(compendium_encoded_df)
```

Comment:
Can probably remove this line

Reply:
I'll leave this for now, since I plan to return to this in the future